### PR TITLE
🌱 Update CoreDNS migration library to v1.0.17

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <kmuldoon@vmware.com>

Cherry pick of https://github.com/kubernetes-sigs/cluster-api/pull/6600 

With custom changes to docs

### CODE REVIEW

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -63,17 +63,17 @@ These diagrams show the relationships between components in a Cluster API releas
...
-| Kubernetes v1.17 | ✓                     |                        |                      |                          |
-| Kubernetes v1.18 | ✓                     | ✓ (only workload)      | ✓ (only workload)    | ✓ (only workload)        |
-| Kubernetes v1.19 | ✓                     | ✓                      | ✓                    | ✓                        |
-| Kubernetes v1.20 | ✓                     | ✓                      | ✓                    | ✓                        |
-| Kubernetes v1.21 | ✓                     | ✓                      | ✓                    | ✓                        |
-| Kubernetes v1.22 | ✓ (only workload)     | ✓                      | ✓                    | ✓                        |
-| Kubernetes v1.23* |                      | ✓                      | ✓                    | ✓                        |
-| Kubernetes v1.24 |                       |                        |                      | ✓                        |
+|                   | CAPI v0.3 (v1alpha3) | CAPI v0.4 (v1alpha4) | CAPI v1.0 (v1beta1) | CAPI v1.1+v1.2 (v1beta1) |
```
Comment (sbueringer) [inline docs/book/src/reference/versions.md:66]: I know I said this before, but it's worth repeating. The GitHub UI diff is so so so bad

Comment (killianmuldoon) [reply docs/book/src/reference/versions.md:66]: Yeah - IDE keeps telling me to reformat these tables - but it's impossible to see what the diff is here ( I think it's probably one tab somewhere)

Comment (sbueringer) [reply docs/book/src/reference/versions.md:66]: I diffed it with Intellij, it's fine :)

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Pull request merged**